docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,504 @@
+to test a "3xx" value and a "4xx" value.
+
+## Assessing coverage
+
+It's very likely that WPT already has some tests for the feature (or at least
+the specification) that you're interesting in testing. In that case, you'll
+have to study what's already been done before starting to write new tests.
+Understanding the design of existing tests will let you avoid duplicating
+effort, and it will also help you integrate your work more logically.
+
+Even if the feature you're testing does *not* have any tests, you should still
+keep these guidelines in mind. Sooner or later, someone else will want to
+extend your work, so you ought to give them a good starting point!
+
+### File names
+
+The names of existing files and folders in the repository can help you find
+tests that are relevant to your work. [This page on the design of
+WPT](../test-suite-design) goes into detail about how files are generally laid
+out in the repository.
+
+Generally speaking, every conformance tests is stored in a subdirectory
+dedicated to the specification it verifies. The structure of these
+subdirectories vary. Some organize tests in directories related to algorithms
+or behaviors. Others have a more "flat" layout, where all tests are listed
+together.
+
+Whatever the case, test authors try to choose names that communicate the
+behavior under test, so you can use them to make an educated guess about where
+your tests should go.
+
+*Example* Imagine you wanted to write a test to verify that headers were made
+immutable by the `Request.error` method defined in [the Fetch
+standard](https://fetch.spec.whatwg.org). Here's the algorithm:
+
+> The static error() method, when invoked, must run these steps:
+>
+> 1. Let *r* be a new [Response](https://fetch.spec.whatwg.org/#response)
+>    object, whose
+>    [response](https://fetch.spec.whatwg.org/#concept-response-response) is a
+>    new [network error](https://fetch.spec.whatwg.org/#concept-network-error).
+> 2. Set *r*'s [headers](https://fetch.spec.whatwg.org/#response-headers) to a
+>    new [Headers](https://fetch.spec.whatwg.org/#headers) object whose
+>    [guard](https://fetch.spec.whatwg.org/#concept-headers-guard) is
+>    "`immutable`".
+> 3. Return *r*.
+
+In order to figure out where to write the test (and whether it's needed at
+all), you can review the contents of the `fetch/` directory in WPT. Here's how
+that looks on a UNIX-like command line:
+
+    $ ls fetch
+    api/                           data-urls/   range/
+    content-encoding/              http-cache/  README.md
+    content-length/                images/      redirect-navigate/
+    content-type/                  metadata/    security/
+    corb/                          META.yml     stale-while-revalidate/
+    cors-rfc1918/                  nosniff/
+    cross-origin-resource-policy/  origin/
+
+This test is for a behavior directly exposed through the API, so we should look
```

Comment:
@annevk can confirm, but I'm pretty sure it's because there's a section in the spec by the name "API". A bunch of WHATWG specs have this but elsewhere it's rare.

Using an example from DOM involving nodes or collections might make for a more pedagogical case.

Reply:
The directory structure in fetch doesn't seem to map to the sections in the spec (even within the `api` dir). I think this example is OK.